core/domain/exp_jobs_one_off.py

```diff
@@ -470,3 +471,115 @@ def map(item):
+            """Finds the value of the key inside all the nested dictionaries
+            in a given dictionary.
+
+            Args:
+                key: str. The key whose value is to be found.
+                dictionary: dict or list. The dictionary or list in which the
+                    key is to be searched.
+
+            Returns:
+                list. The values of the key in the given dictionary.
+            """
+            if isinstance(dictionary, list):
+                for d in dictionary:
+                    if isinstance(d, (dict, list)):
+                        for result in find(key, d):
+                            yield result
+            else:
+                for k, v in dictionary.iteritems():
+                    if k == key:
+                        yield v
+                    elif isinstance(v, dict):
+                        for result in find(key, v):
+                            yield result
+                    elif isinstance(v, list):
+                        for d in v:
+                            if isinstance(d, (dict, list)):
+                                for result in find(key, d):
+                                    yield result
+
+        if item.deleted:
+            return
+
+        exploration = exp_services.get_exploration_from_model(item)
+
+        allowed_parent_list = {
+            'p': ['blockquote', 'div', 'pre', '[document]'],
+            'b': ['i', 'li', 'p', 'pre'],
+            'br': ['b', 'i', 'li', 'p'],
+            'div': ['blockquote'],
+            'i': ['b', 'li', 'p', 'pre'],
+            'li': ['ol', 'ul'],
+            'ol': ['blockquote', 'li', 'pre', 'div', '[document]'],
+            'ul': ['blockquote', 'li', 'pre', 'div', '[document]'],
+            'pre': ['blockquote', '[document]'],
+            'blockquote': ['blockquote', '[document]'],
+            'oppia-noninteractive-link': ['b', 'i', 'li', 'p', 'pre'],
+            'oppia-noninteractive-math': ['b', 'i', 'li', 'p', 'pre'],
+            'oppia-noninteractive-image': ['li', 'p', 'pre'],
+            'oppia-noninteractive-collapsible': ['li', 'p', 'pre'],
+            'oppia-noninteractive-tabs': ['li', 'p', 'pre'],
+            'oppia-noninteractive-video': ['li', 'p', 'pre']
+        }
+
+        allowed_tag_list = [
+            'p', 'b', 'br', 'div', 'i', 'li', 'ol', 'ul', 'pre', 'blockquote',
+            'oppia-noninteractive-link',
+            'oppia-noninteractive-math',
+            'oppia-noninteractive-image',
+            'oppia-noninteractive-collapsible',
+            'oppia-noninteractive-tabs',
+            'oppia-noninteractive-video'
+        ]
+
+        err_list = {}
+
+        for state in exploration.states.itervalues():
+            result = find('html', state.to_dict())
+
+            for html_data in result:
+                html_data = html_data.encode('utf-8')
+                soup = BeautifulSoup(html_data, 'html.parser')
+
+                # For invalid tags.
```

Comment:
I would expand this and the comment below, so other people can better understand what is going on. Perhaps, "checking for tags that are not allowed by the RTE."

Reply:
Done!